### PR TITLE
fix: command LspStop can receive server name

### LINKED ISSUE #3362: `:LspStop config-name` (as opposed to `:LspStop client-id`)
### Description

Apologies if there has already been an issue for this - I couldn't find anything out there that seemed similar. 

There are some occasions where I would like to be able to stop a specific LSP (like `eslint`), but leave all other language servers running on a buffer (`volar`, `tailwindcss`, etc.)

Currently, the only way that I know of to disable LSPs on a buffer is with `:LspStop`, but this stops all language servers.

I think it would be a nice feature to modify this to accept LSP server names like `:LspStop eslint` so we can stop specific LSPs that are attached to buffer and leave all the others attached.

### PR DESCRIPTION
**Pull request by glepnir**
Receive server name as command argument then close this server if find it.

Fix #3362

### CODE REVIEW

File: plugin/lspconfig.lua [hunk 1]
```diff
@@ -110,20 +110,22 @@ end, {
...
     end
   end
 
   if not server_id then
     local servers_on_buffer = require('lspconfig.util').get_lsp_clients { bufnr = current_buf }
     for _, client in ipairs(servers_on_buffer) do
-      if client.attached_buffers[current_buf] then
+      if client.attached_buffers[current_buf] and (server_name and (server_name == client.config.name) or true) then
         client.stop(force)
       end
```
Comment (justinmk) [inline plugin/lspconfig.lua:130]: if server_name is provided but not found, we should show a `config "foo" not found` message to indicate that. silent failure is usually a bad UX.

Comment (glepnir) [reply plugin/lspconfig.lua:130]: make sense

File: plugin/lspconfig.lua [hunk 2]
```diff
@@ -110,23 +110,31 @@ end, {
...
     for _, client in ipairs(servers_on_buffer) do
-      if client.attached_buffers[current_buf] then
+      if client.attached_buffers[current_buf] and (server_name and (server_name == client.config.name) or true) then
         client.stop(force)
+        found = true
       end
     end
+
+    if server_name and not found then
+      vim.notify(('LSP server "%s" not found'):format(server_name), vim.log.levels.WARN)
```
Comment (justinmk) [inline plugin/lspconfig.lua:136]: Is it really the server or just the config name? "LSP server" seems misleading, it sounds like the server program name could not be found?
```suggestion
      vim.notify(('nvim-lspconfig: config "%s" not found'):format(server_name), vim.log.levels.WARN)
```

**Review by justinmk — APPROVED** [review]
Nice!

**Comment by justinmk** [comment]
Oh, we forgot to update the docs :)

**Pull request merged**